### PR TITLE
Update quest documentation

### PR DESCRIPTION
**Pull request by gabigarc03**
Updated `quests_state.h` and `quests_structs.h` with more accurate documentation that that reflects what functions are actually doing, what the structs are and their components, and uses proper language (e.g. no longer refers to a quest marked "failed" as marked "started") as described in issue #889.

### CODE REVIEW

File: include/quests/quests_state.h [hunk 1]
```diff
@@ -167,43 +169,40 @@ int fail_quest(quest_t *quest);
...
+ * - 0 if quest is incomplete
+ * - 1 if quest is complete
  */
 int is_quest_completed(quest_t *quest);
 
-/* Checks the status of a quest
+/* Checks a quest's status.
  *
  * Parameter:
  * - quest: pointer to quest_t
```
Comment (ddelgado60) [inline include/quests/quests_state.h:186]: Minor comment, but should this be quest instead of quest_t?

File: include/quests/quests_state.h [hunk 2]
```diff
@@ -120,19 +121,19 @@ int achievement_free(achievement_t *achievement);
...
  * - SUCCESS 
  * - FAILURE
  */
 int add_achievement_to_quest(quest_t *quest, achievement_t *achievement_to_add);
 
-/* Starts a quest
+/* Updates a quest's status to started
  *
  * Parameter:
  * - quest: pointer to quest_t to be started
```
Comment (ddelgado60) [inline include/quests/quests_state.h:139]: Minor comment, but should this be quest instead of quest_t?

File: include/quests/quests_state.h [hunk 3]
```diff
@@ -120,19 +121,19 @@ int achievement_free(achievement_t *achievement);
...
 int quest_free(quest_t * quest);
 
-/* Adds an achievement to a quest
+/* Adds an achievement to the end of a quest's achievement list
  *
- * Parameter:
- * - quest: pointer to quest_t 
- * - achievement_to_add: pointer to achievement_t to add to list
+ * Parameters:
+ * - quest: pointer to a quest_t
```
Comment (ddelgado60) [inline include/quests/quests_state.h:127]: Minor comment, but should this be quest instead of quest_t?

**Review by ddelgado60 — CHANGES_REQUESTED** [review]
Looks good! Great work on updating the documentation from last year. I had a quick comment: when the documentation has an achievement or a quest as input, I'm not sure if it should be like quest: a pointer to quest or quest: a pointer to quest_t. I think it may be the former, so perhaps you could take a look. I only commented directly below on a few.

**Review by ddelgado60 — APPROVED** [review]
Looks great! Thanks for reviewing my comments.

**Review by lilyehsani — CHANGES_REQUESTED** [review]
Great job! Added a few minor comments

File: include/quests/quests_state.h [hunk 4]
```diff
@@ -167,43 +169,40 @@ int fail_quest(quest_t *quest);
  */
 int complete_achievement(quest_t *quest, item_t *item_collected, npc_t *npc_met);
 
-/* Checks if the quest is completed
+/* Checks if a quest is completed
  * 
- * Paramter:
+ * Parameter:
  * - pointer to the quest
```
[the remark this answers is not in the document]
Comment (gabigarc03) [reply include/quests/quests_state.h]: Fixed! Thank you so much.

File: include/quests/quests_state.h [hunk 5]
```diff
@@ -109,54 +109,56 @@ int mission_free(mission_t *mission);
...
  * - FAILURE
  */
 int add_achievement_to_quest(quest_t *quest, achievement_t *achievement_to_add);
 
-/* Starts a quest
+/* Updates a quest's status to started
  *
  * Parameter:
- * - quest: pointer to quest_t to be started
+ * - quest: pointer to quest to be started
```
[the remark this answers is not in the document]
Comment (gabigarc03) [reply include/quests/quests_state.h]: Fixed! Thank you so much.

**Review by lilyehsani — APPROVED** [review]
Looks great! Approved.

**Comment by gabigarc03** [comment]
`quests_state.h` and `quests_struct.h` now have updated documentation that accurately reflects the structs and functions within them.

**Comment by lilyehsani** [comment]
Issue Score: ✔️++

Comments:
Great work overall! The summary/closing and peer review comments of this PR are sufficiently thorough.

**Pull request merged**